readthedocs/search/utils.py

```diff
@@ -5,36 +5,26 @@
 from django_elasticsearch_dsl.apps import DEDConfig
 from django_elasticsearch_dsl.registries import registry
 
-from readthedocs.projects.models import HTMLFile
+from readthedocs.search.documents import PageDocument
 
 log = structlog.get_logger(__name__)
 
 
-def index_new_files(model, version, build):
-    """Index new files from the version into the search index."""
+def index_objects(document, objects, index_name=None):
+    # If a search index name is provided, we need to temporarily change
+    # the index name of the document.
+    old_index_name = document._index._name
+    if index_name:
+        document._index._name = index_name
 
-    log.bind(
-        project_slug=version.project.slug,
-        version_slug=version.slug,
-    )
-
-    if not DEDConfig.autosync_enabled():
-        log.info("Autosync disabled. Skipping indexing into the search index.")
-        return
+    document().update(objects)
```

Comment:
Don't we need to check if `ELASTICSEARCH_DSL_AUTOSYNC` is enabled here before executing this function? I understand that the check we had previously, `DEDConfig.autosync_enabled()`, did this check.